lambda_function.py

```diff
@@ -1,132 +1,224 @@
+    Abstract class for data fetching
+    """
+    @abstractmethod
+    def fetch_data(self):
+        """
+        Method for fetching data
+        """
+        pass
 
 
-def request_data_from_notion():
-    notion_url = "https://energetic-tuberose-21a.notion.site/api/v3/loadCachedPageChunk"
-    response = requests.post(
-        notion_url, 
-        json={
-                "page": {"id": NOTION_PAGE_ID},
+class DataParser(ABC):
+    """
+    Abstract class for parsing data 
+    """
+    @abstractmethod
+    def parse_response(self, data):
+        """
+        Method for parsing data
+        """
+        pass
+
+
+class NotionBaseClient(DataFetcher, DataParser, ABC):
+    """
+    Notion abstract class
+    """
+    pass
+
+class NotionClient(NotionBaseClient):
+    """
+    Implementation of NotionData
+    """
+    def __init__(self, notion_page_id, notion_url):
+        self.notion_page_id = notion_page_id
+        self.notion_url = notion_url
+
+    def fetch_data(self):
+        response = requests.post(
+            self.notion_url,
+            json={
+                "page": {"id": self.notion_page_id},
                 "limit": 30,
-                "cursor":{"stack": []},
-                "chunkNumber":0 ,
-                "verticalColumns": False}
-    )
-    response_data= response.json()
-    data=response_data.get("recordMap", {}).get("block", {})
-    return data
-
-def parse_response(data):
-    page = []
-    for _, data_value in data.items():
-        value = data_value.get('value')
-        value_type = value.get('type')
-        title = value.get('properties', {}).get('title', [['']])
-        text = parse_text(title)
-        
-        if not text:
-            break
-        
-        text = format_text(text, value_type)
-        if text:
-            page.append(text)
-    page.append(AUTOMATED_MESSAGE)
-    page_string = '\n'.join(page)
-    return page_string
-
-def print_data(page):
-    """
-    For debugging
-    """
-    for line in page:
-        print(line)
-
-
-def post_to_slack(text, thread_id):
-    try:
-        result = slack_client.chat_postMessage(
-            channel=CHANNEL_ID,
-            text=text,
-            thread_ts=thread_id
+                "cursor": {"stack": []},
+                "chunkNumber": 0,
+                "verticalColumns": False
+            }
         )
-        logger.info(result)
+        response_data = response.json()
+        data = response_data.get("recordMap", {}).get("block", {})
+        return data
+
+    def parse_text(self, title):
+        """
+        Method for parsing text
+        """
+        text_list = []
+        for text_data in title:
+            if len(text_data) == 1:
+                text_list.append(text_data[0])
+            elif text_data[1][0] and text_data[1][0][0] == 'a':
+                text_list.append(text_data[0])
+        complete_text = ''.join(text_list)
+        return complete_text
+
+    def format_text(self, text, value_type):
```

Comment:
Is it a good idea to keep the data fetching logic from a third party source and the parsing logic in the same class?